File: test/fixtures/ts-mini/src/store.ts
import { AbstractStore } from "./base";
import { isExpired, Session, User } from "./models";

/** Keeps sessions in a Map; refresh extends their TTL. */
export class SessionStore extends AbstractStore {
  private sessions = new Map<string, Session>();

  get(token: string): Session | undefined {
    return this.sessions.get(token);
  }

  put(session: Session): void {
    this.sessions.set(session.token, session);
  }

  /** Extend a session's lifetime; drops expired sessions. */
  refresh(token: string, extraTtl = 3600): Session | undefined {
    const session = this.get(token);
    if (!session) return undefined;
    if (isExpired(session)) {
      this.sessions.delete(token);
      return undefined;
    }
    session.ttl += extraTtl;
    this.put(session);
    return session;
  }

  create(user: User, token: string): Session {
    const session: Session = { token, user, ttl: 3600 };
    this.put(session);
    return session;
  }
}

File: test/fixtures/py-mini/app/store.py
"""In-memory session store."""
from typing import Dict, Optional

from .base import BaseStore
from .models import Session, User


class SessionStore(BaseStore):
    """Keeps sessions in a dict; refresh extends their TTL."""

    def __init__(self) -> None:
        self._sessions: Dict[str, Session] = {}

    def get(self, token: str) -> Optional[Session]:
        return self._sessions.get(token)

    def put(self, session: Session) -> None:
        self._sessions[session.token] = session

    def refresh(self, token: str, extra_ttl: int = 3600) -> Optional[Session]:
        """Extend a session's lifetime; drops expired sessions."""
        session = self.get(token)
        if session is None:
            return None
        if session.is_expired():
            self._sessions.pop(token, None)
            return None
        session.ttl += extra_ttl
        self.put(session)
        return session

    def create(self, user: User, token: str) -> Session:
        session = Session(token=token, user=user)
        self.put(session)
        return session
